code/game/objects/items/hand_item.dm

```diff
@@ -62,13 +62,26 @@
 /obj/item/slapper/parry
 	desc = "This is how real men win fights."
 	force = 5
+	flags = DROPDEL | ABSTRACT | NODROP
 	attack_verb = list("slapped", "backhanded", "smacked", "discombobulated")
 	table_smacks_left = 10 //Much more smackitude
 
 /obj/item/slapper/parry/Initialize(mapload)
 	AddComponent(/datum/component/parry, _stamina_constant = 2, _stamina_coefficient = 0.5, _parryable_attack_types = NON_PROJECTILE_ATTACKS, _parry_cooldown = (1 / 3) SECONDS) //75% uptime
+	if(isliving(loc))
+		var/mob/owner = loc
+		RegisterSignal(owner, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(drop_slapper), override = TRUE)
+		RegisterSignal(owner, COMSIG_MOB_WEAPON_APPEARS, PROC_REF(drop_slapper), override = TRUE)
```

Comment:
tbh, this has been used in so many places. It really should be made into a component

Reply:
excuse me, what exactly should be a component?
It is currently in 3 items in game(maybe 4) and they add signals to different entities.
Changeling abilities add signals from action to a host, so when you press drop button, action controls removing armblade/tentackle/shield, if you dont have the action, well, you cant remove your armblade.
CQC slapper adds signals from slapper itself because you should always be able to remove it no matter what hapenned to your CQC.

Reply:
Fair point, but I still don't like how theres tons of these things now that have willingly drop on them that are used in so many different ways. I feel like they should be standardized.

Reply:
Yeah, until it's used more, imo it is fine to not have it as a component